ts/features/messages/saga/handleUpsertMessageStatusAttributes.ts

```diff
@@ -0,0 +1,230 @@
+) {
+  do {
+    const currentEntryToProcess = yield* select(
+      nextQueuedMessageDataUncachedSelector
+    );
+    if (!currentEntryToProcess) {
+      const userFeedback = I18n.t("messages.operations.generic.success");
+      yield* put(
+        resetMessageArchivingAction({ type: "success", reason: userFeedback })
+      );
+      return;
+    }
+
+    const category: MessageListCategory = currentEntryToProcess.archiving
+      ? "INBOX"
+      : "ARCHIVE";
+    const message = yield* select(
+      paginatedMessageFromIdForCategorySelector,
+      currentEntryToProcess.messageId,
+      category
+    );
+    if (!message) {
+      yield* put(
+        removeScheduledMessageArchivingAction({
+          fromInboxToArchive: currentEntryToProcess.archiving,
+          messageId: currentEntryToProcess.messageId
+        })
+      );
+      continue;
+    }
+
+    const upsertMessageStatusAttributesAction =
+      upsertMessageStatusAttributes.request({
+        message,
+        update: {
+          isArchived: currentEntryToProcess.archiving,
+          tag: "archiving"
+        }
+      });
+    yield* put(upsertMessageStatusAttributesAction);
+
+    const outputAction = yield* take([
+      upsertMessageStatusAttributes.success,
+      upsertMessageStatusAttributes.failure,
+      resetMessageArchivingAction
+    ]);
+
+    if (isActionOf(upsertMessageStatusAttributes.success, outputAction)) {
+      yield* put(
+        removeScheduledMessageArchivingAction({
+          fromInboxToArchive: currentEntryToProcess.archiving,
+          messageId: currentEntryToProcess.messageId
+        })
+      );
+    } else {
+      if (isActionOf(upsertMessageStatusAttributes.failure, outputAction)) {
+        const userFeedback = I18n.t("messages.operations.generic.failure");
+        yield* put(
+          interruptMessageArchivingProcessingAction({
+            type: "error",
+            reason: userFeedback
+          })
+        );
+      }
+      return;
+    }
+  } while (true);
+}
+
+// Be aware that this saga is execute with a takeEvery, in order to remain
+// compatible with the old messages home way of archiving messages
+export function* raceUpsertMessageStatusAttributes(
+  putMessage: BackendClient["upsertMessageStatusAttributes"],
+  action: ActionType<typeof upsertMessageStatusAttributes.request>
+) {
+  yield* race({
+    task: call(handleUpsertMessageStatusAttributes, putMessage, action),
+    cancel: take(resetMessageArchivingAction)
+  });
+}
+
+export function* handleUpsertMessageStatusAttributes(
+  putMessage: BackendClient["upsertMessageStatusAttributes"],
+  action: ActionType<typeof upsertMessageStatusAttributes.request>
+) {
+  try {
+    const body = validatePayload(action.payload);
+    const response = (yield* call(
+      withRefreshApiCall,
+      putMessage({ id: action.payload.message.id, body }),
+      action
+    )) as unknown as SagaCallReturnType<typeof putMessage>;
```

Comment:
```suggestion
    )) as SagaCallReturnType<typeof putMessage>;
```
we can remove the unknown casting here :p

Reply:
Nice catch, done